### PR TITLE
Scheduler upgrades

### PR DESCRIPTION
**Pull request by detiber**

### CODE REVIEW

**Comment by detiber** [comment]
aos-ci-test

**Comment by detiber** [comment]
@dgoodwin ptal

@sdodson once this gets a +1 I can create the backport PRs.

**Comment by detiber** [comment]
aos-ci-test

File: playbooks/common/openshift-cluster/upgrades/upgrade_scheduler.yml [hunk 1]
```diff
@@ -0,0 +1,160 @@
+---
+# Upgrade predicates
+- vars:
+    prev_predicates: "{{ lookup('openshift_master_facts_default_predicates', short_version=openshift_upgrade_min, deployment_type=openshift_deployment_type) }}"
+    prev_predicates_no_region: "{{ lookup('openshift_master_facts_default_predicates', short_version=openshift_upgrade_min, deployment_type=openshift_deployment_type, regions_enabled=False) }}"
+    default_predicates_no_region: "{{ lookup('openshift_master_facts_default_predicates', regions_enabled=False) }}"
+    older_predicates:
```
Comment (dgoodwin) [inline playbooks/common/openshift-cluster/upgrades/upgrade_scheduler.yml]: What is older predicates?

**Comment by detiber** [comment]
aos-ci-test

File: playbooks/common/openshift-cluster/upgrades/upgrade_scheduler.yml [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    - - name: MatchNodeSelector
+      - name: PodFitsResources
+      - name: PodFitsPorts
+      - name: NoDiskConflict
+  block:
+
+  # Handle case where openshift_master_predicates is defined
+  - block:
+    - debug:
+        msg: "WARNING: openshift_master_scheduler_predicates is set to defaults from an earlier release of OpenShift current defaults are: {{ openshift_master_scheduler_default_predicates }}"
```
Comment (detiber) [inline playbooks/common/openshift-cluster/upgrades/upgrade_scheduler.yml]: I must have missed it, what I thought I saw was something that tested the result of a particular action and issued a warning.

File: playbooks/common/openshift-cluster/upgrades/upgrade_scheduler.yml [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+- vars:
+    scheduler_config:
+      kind: Policy
+      apiVersion: v1
+      predicates: "{{ openshift_upgrade_scheduler_predicates
+                      | default(openshift_master_scheduler_current_predicates) }}"
+      priorities: "{{ openshift_upgrade_scheduler_priorities
+                      | default(openshift_master_scheduler_current_priorities) }}"
+  block:
+  # TODO: perform upgrade
```
Comment (detiber) [inline playbooks/common/openshift-cluster/upgrades/upgrade_scheduler.yml]: of course, what is a PR without a little cruft :)

File: playbooks/common/openshift-cluster/upgrades/upgrade_scheduler.yml (hunk 2 above)
Comment (detiber) [inline playbooks/common/openshift-cluster/upgrades/upgrade_scheduler.yml]: To elaborate further, I was worried about moving all of the tests that do warnings into post_control_plane, and I was also worried about depending on a persisted variable to be present between the playbooks.

**Comment by detiber** [comment]
aos-ci-test

**Comment by detiber** [comment]
@sdodson ptal

backport PRs:
- 3.4: https://github.com/openshift/openshift-ansible/pull/2930
- 3.3: https://github.com/openshift/openshift-ansible/pull/2931
- 3.2: https://github.com/openshift/openshift-ansible/pull/2932

File: roles/openshift_master_facts/test/openshift_master_facts_default_predicates_tests.py [hunk 4]
```diff
@@ -91,39 +146,70 @@ def check_defaults(self, release, deployment_type, default_predicates,
...
+        facts['openshift_version'] = version
+        facts['openshift']['common']['deployment_type'] = deployment_type
+        results = self.lookup.run(None, variables=facts,
+                                  regions_enabled=regions_enabled)
+        if regions_enabled:
+            assert_equal(results, default_predicates + [REGION_PREDICATE])
+        else:
+            assert_equal(results, default_predicates)
+
+    def check_defaults_override_vars(self, release, deployment_type,
```
Comment (rhcarvalho) [inline roles/openshift_master_facts/test/openshift_master_facts_default_predicates_tests.py]: @detiber, this and the other method with the same name in the other test file are deadcode, not called by anything.

Is there a test method missing or shall I just delete these methods?

Comment (rhcarvalho) [inline roles/openshift_master_facts/test/openshift_master_facts_default_predicates_tests.py]: FYI I'm rewriting these tests using pytest :-)

Comment (detiber) [inline roles/openshift_master_facts/test/openshift_master_facts_default_predicates_tests.py]: They can be killed. Looking forward to seeing the purest version

**Pull request merged**